lib/ea_restaurant_data_loader_web/controllers/oauth2_controller.ex

```diff
@@ -0,0 +1,13 @@
+defmodule EaRestaurantDataLoaderWeb.Controllers.Oauth2Controller do
+  use EaRestaurantDataLoaderWeb, :controller
+  alias EaRestaurantDataLoader.Lib.Utils.Oauth2Util
+  alias EaRestaurantDataLoader.Lib.Services.Oauth2Service
+
+  def login(conn, _params) do
+    [basic_auth | _] = conn |> get_req_header("authorization")
```

Comment:
What would happen if the header does not exist?

Reply:
It will fail pattern Matching raising MatchError and will be catch by the try/rescue.